signbank/dictionary/adminviews.py

```diff
@@ -698,6 +698,48 @@ def serialize_glosses_csv(dataset, queryset):
     return render_to_csv_response(queryset)
 
 
+def get_video_details(request):
+    """ This function will return a csv with details of published videos. No filtering the page would affect here. """
+
+    if not request.user.has_perm('dictionary.export_csv'):
+        msg = _("You do not have permissions to export to CSV.")
+        messages.error(request, msg)
+        raise PermissionDenied(msg)
+
+    # Create the HttpResponse object with the appropriate CSV header.
+    response = HttpResponse(content_type='text/csv; charset=utf-8')
+    response['Content-Disposition'] = 'attachment; filename="dictionary-export.csv"'
+
+    writer = csv.writer(response)
+
+    csv_queryset =GlossVideo.objects.filter(is_public=True)
+
+    # We want to manually set which fields to export here
+    fieldnames = ['id', 'videofile', 'version', 'gloss_id', 'dataset', 'title', 'video_type_id']
+    fields = [GlossVideo._meta.get_field(fieldname) for fieldname in fieldnames]
+
+    # Defines the headings for the file. Signbank ID and Dataset are set first.
+    header = [f.verbose_name for f in fields]
+
+    writer.writerow(header)
+
+    for video in csv_queryset:
+        row = list()
+        # Add data from each field.
+        for f in fields:
+            value = str(getattr(video, f.name))
+
+            # If the value contains ';', put it in quotes.
+            if value and ";" in value:
+                row.append('"{}"'.format(value))
```

Comment:
Does Python not handle escaping characters for you?

Reply:
Sorry to butt in, but - is it safe to escape characters in a CSV formatted file? Will they not appear with the '\' or whatever in the target output?  Or does that not matter in this case?